cdap-common/src/main/java/io/cdap/cdap/common/lang/jar/BundleJarUtil.java

```diff
@@ -182,6 +185,13 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
    * @throws IOException If failed to expand the jar
    */
   public static File prepareClassLoaderFolder(Location jarLocation, File destinationFolder) throws IOException {
+    // If jarLocation is a local location and its a directory assume its already unpacked
+    URI uri = jarLocation.toURI();
+    if ("file".equals(uri.getScheme()) && jarLocation.isDirectory()) {
```

Comment:
Please update javadoc to say it's accepting folders and not only jars

Reply:
done